### PR TITLE
Add and fix tests for multi-dispatching methods

### PR DESCRIPTION
**Pull request by vrurg**
In support for rakudo/rakudo#3977.

### CODE REVIEW

File: S12-methods/multi.t [hunk 1]
```diff
@@ -38,10 +38,8 @@ is($foo.bar("Hello"), 'Foo.bar() called with Str : Hello', '... multi-method dis
 is($foo.bar(5), 'Foo.bar() called with Int : 5', '... multi-method dispatched on Int');
 is($foo.bar(4.2), 'Foo.bar() called with Numeric : 4.2', '... multi-method dispatched on Numeric');
 
-# https://github.com/Raku/old-issue-tracker/issues/1019
-#?rakudo todo 'RT #66006'
-try { EVAL '$foo.baz()' };
-ok ~$! ~~ /:i argument[s?]/, 'Call with wrong number of args should complain about args';
+throws-like { $foo.baz() }, X::Multi::NoMatch,
+    "Call with wrong nnumber of args results in no matching candidate exception";
```
Comment (lizmat) [inline S12-methods/multi.t:42]: s/nnumber/number

File: S12-methods/multi.t [hunk 2]
```diff
@@ -115,7 +113,7 @@ is Bar.new.a("not an Int"), 'Any-method in Foo';
     is $m.d( '7' ), 'string',  'dispatch to other role';
     is $m.d( 1.2 ), 'any',     'dispatch to the class with the roles';
 
-    my @multi_method = $m.^methods.grep({ ~$_ eq 'd' });
+    my @multi_method = $m.^methods.grep({ .gist eq 'd' });
```
Comment (lizmat) [inline S12-methods/multi.t:116]: Why not `.name` instead of `.gist` ?

**Review by lizmat — CHANGES_REQUESTED** [review]
Two small nits, otherwise ok for merging :-)

File: S12-methods/multi.t (hunk 2 above)
Comment (vrurg) [reply S12-methods/multi.t:116]: I was to tired to check it deeper, but basically you back up my yesterday thoughts. Will commit the change later today.

**Pull request merged**